### DISCUSSION TITLE
Welcome to TonexOneController Discussions!

### DISCUSSION THREAD

**Discussion by Builty** [Builty/TonexOneController#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by waltiti17** [comment]
Hey Greg Already left some comments on your YouTube videos and facebook post.
First thanks a lot, a lot.

Second, some suggestions.

1) Is possible to handle 2 tonex one units, not at the same time, maybe with a physical switch or software switch in the TonexOneController software?  

2) is  possible to use on the big Tonex Pedal, I think is a complete diferent unit, but maybe. 

Thanks a lot again

**Comment by dxmadrid** [comment]
Hello .

A video tutorial of the start-up would be nice, for those of us who are not familiar with Arduino. 
Thank you

**Reply by Builty** [reply]
Hi. Do you mean the process of putting the code onto the board, or the whole code development part?
First one I plan to do soon and is not difficult. The second one is far more complicated.

**Reply by dxmadrid** [reply]
Todo ok mi esp32s3zero funcionando sin problema.

**Reply by Builty** [reply]
Great

**Comment by denysetiaji** [comment]
hello greg, i dont undeestand. for tonex no display. i just create that board and 5v adapter or need doing some stuff like coding?

**Reply by Builty** [reply]
Hi. There are two steps needed.
1. Do the wiring, as per https://github.com/Builty/TonexOneController/blob/main/HardwarePlatforms.md#waveshare_zero
2. Upload the firmware, as per https://github.com/Builty/TonexOneController/blob/main/FirmwareUploading.md

You don't need to write any code or anything special like that. But you do need to put my code onto the PCB, which is what step 2 does.

This might help also:
https://www.youtube.com/watch?v=qkOs5gk3bcQ

**Reply by denysetiaji** [reply]
thank you for the respon, but is there any chance for using this to boss gx 10?

**Reply by Builty** [reply]
Hi. Sorry I don't know anything about the Boss gx10.
I can see it already has footswitches, and can use bluetooth also. I don't know what you would gain from adding this controller.

**Reply by denysetiaji** [reply]
its need usb midi like host midi i think, because i need to using pc or phone and app for using usb midi to wireless midi controller. bluetooth needs separate item to buy boss bt-dual.i just need mvave chcocolate for PC command to change preset

**Reply by Builty** [reply]
It's probably possible to change the code to do this. Replacing the Tonex host code with USB Midi code

**Comment by Zivot76** [comment]
Hi everyone! First of all thanks for this amazing project. I apologise if this question has already been asked: is it possible to connect 2 m-vave chocolate via bluetooth at the same time? I have 2, non plus, and I absolutely need all 8 footswitches to use the tonex one with my hx stomp and my analogue pedals connected to a loop switcher. Thanks to anyone who would like to answer me.

**Comment by Builty** [comment]
Hi. At the moment, only 1 footswitch pedal can be connected via Bluetooth. 
You can however use one via Bluetooth and one via wired Midi, but it needs more hardware.

I might be able to suport more than one over bluetooth, but it would take me some time to try it. Right now I am working on supporting 8 wired footswitches, and soon trying to USB midi working.

**Reply by Zivot76** [reply]
Thank you for reply, l will wait your upgrades and continue to enjoy this nice community. :))

**Reply by rmenes** [reply]
> Thank you for reply, l will wait your upgrades and continue to enjoy this nice community. :))

I was able to use one chocolate via bluetooth and one via midi as Builty said. I used one chocolate to send PC commands to change presets, and another to send CC commands to turn on and off the effects.
The blue pedal is the tonex controller and I conected MIDI using a TRS cable instead of using DIN connectors.


![Sem título-1](https://github.com/user-attachments/assets/92bf5647-d304-478b-b61a-550eb5d55cf5)

**Comment by Zivot76** [comment]
Hi everybody. I am a owner of the CME U6Midi Pro (3 midi in, 3 midi out) which connects through 2 midi input my 2 m-vave chocolate and send midi commands to 2 Midi Looper Switcher (G-Lab M4L Midi 4xLoop) and to the HX-Stomp. 
I have now purchased the Tonex One pedal to use with my HX Stomp. 
Today I wrote to CME if any of their products that can receive midi data ('Program Change' type commands) from the U6Midi Pro and transmit it to the Tonex One allowing selection of the 20 available presets. I looked at their product CME "**H2MIDI Pro**", and I asked them if it is capable of doing this?
They replyed me soon, this is what they wrote:

Dear Antonio,
Thanks for your email.
The H2MIDI Pro can work with Tonex One via USB. But do you still have free MIDI port on U6MIDI Pro? If yes, you can connect the H2 to U6 via MIDI cable.
Best Regards
Zhao Yitian / 赵易天
CEO of CME - Always One Step Ahead
Website: www.cme-pro.com

What do u think? Anyone of you have some experience about it? What do u suggest me?
Thanks very much in advance for anyone would help me. 
Antonio

**Comment by Zivot76** [comment]
Hi everybody.  I just finished assembling my waveshare zero without the display and in the "Bluetooth Client mode" with banks it works properly. Maybe I didn't understand it, but with chocolate in "advanced custom mode" it is possible to use "Bluetooth Server mode" so to be able to send Program change messages 0 to 19? What do I have to do to use the PC messages? 
Thanks very much in advance for anyone would help me.
Antonio

**Reply by Builty** [reply]
Hi. I don't think the Chocolate pedal would support Client mode itself. That is a function that is more often found on phones, PCs etc.

Program change messages from 0 to 19 are exactly what the Chocolate pedal sends in normal operation. 
In Advanced custom mode, you could send Control Change messages (like enabling effects) but that is not needed for preset changing.

**Reply by Zivot76** [reply]
Tnx for the answer.. so, if I understand right, there is no way to change presets sending Program Changes messages trough chocolate in 'advanced custom mode', and this is right what I need. Anyway, I have to look for an other solution.

**Reply by Builty** [reply]
I'm not super familiar with the advanced side of the Chocolate. If it could send control change messages, there are preset up/down messages there.
I don't know if it can send program change from advanced mode.

**Comment by pwaugh** [comment]
Just downloaded the release zip files for 4.3B Touch, and am unable to unzip them on my MAC.  They are in an unsupported format.

**Reply by Builty** [reply]
I'm not a Mac user myself, but other Mac users have used the web method, so were able to unzip them.
I'm not sure what to suggest. I can post the unzipped binary files here if that helps.

**Reply by pwaugh** [reply]
I figured it out.  I was trying to save the files with the link, but that wasn't a zip file.  :).    I just cloned the repository and am set now.

**Comment by DdMan1301** [comment]
Hi! Thank you so much for the great support!!!  Is it possible to use a chocolate bar to simply switch presets back and forth with two buttons, from 1 to 20?  I couldn't set up this option, I tried using the midi command numbers from big tonex, but it didn't help me. If possible, from which team it is. I've tried 86/87 and 89/90, in CC mode and PC mode. Or are such features not enabled yet? I've seen similar questions on the forum, but I haven't received an answer.

**Comment by Builty** [comment]
Hi. Midi CC 86/87 are implemented, but it was done recently. Are you running the latest V1.0.9.2?
There is also an enable in the supporting CC commands over Midi, which is disabled by default.

With this setting (in web config) enabled, 86/87 should work.
Bank on 89/90 are not currently implemented, as the Tonex One doesn't have internal banks like the big Tonex does.

**Reply by DdMan1301** [reply]
Thanks a lot for the answer, I solved the issue by enabling Custom Midi CC command. Thank you again for the wonderful work!

**Comment by macbang707** [comment]
I'm truly delighted to have discovered this amazing community.
Thanks to your incredible work, I've been enjoying the ToneX ONE controller with the 4.3-inch display to great satisfaction.

This site has also been gaining a lot of attention and popularity in the community I’m active in.
There’s one small but frequently mentioned request in that community, which I'd like to kindly share here.

While the 4.3-inch version is excellent, it feels a bit too large for some setups — including mine.
If it’s at all possible, would you consider offering a version optimized for a 3.5-inch display as well?
I believe not only myself, but many others would be genuinely excited and grateful to see a version tailored for the smaller screen.

Thank you sincerely for sharing such valuable work. You have my full support and deepest appreciation.

**Reply by Builty** [reply]
Hi. Thank you for the kind words.
I've actually had a number of requests for a 3.5" display. I actually already have a Waveshare 3.5" (https://www.waveshare.com/esp32-s3-touch-lcd-3.5.htm) , and also another model on its way (https://www.aliexpress.com/item/1005007593889279.html.)

I think it will take a few weeks for me to get to it, but yes this should be possible.

**Reply by Builty** [reply]
I have the Waveshare 3.5" and the JC3248W 3.5" boards both working now. 

<img width="1311" height="749" alt="image" src="https://github.com/user-attachments/assets/3a038456-904f-4c3c-9169-04f38ec134a8" />

**Reply by macbang707** [reply]
> 저는 Waveshare 3.5인치와 JC3248W 3.5인치 보드를 둘 다 사용하고 있는데, 둘 다 잘 작동합니다.
> 
> <img alt="영상" width="1311" height="749" src="https://private-user-images.githubusercontent.com/23299822/470217355-3a038456-904f-4c3c-9169-04f38ec134a8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMzNTcxNjIsIm5iZiI6MTc1MzM1Njg2MiwicGF0aCI6Ii8yMzI5OTgyMi80NzAyMTczNTUtM2EwMzg0NTYtOTA0Zi00YzNjLTkxNjktMDRmMzhlYzEzNGE4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzI0VDExMzQyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg2MDA0OTU3ZDljMzUzMWRkOTRhYjEzMDZhODQyYWJjNzY3NGQ3ZmJkOWZiMTM3OTFlMTkyMGRlNTczZGMzOWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xwoPld9Xfcw7y9J7-sZib2HO8w2ObYK6XyF1Pav0DVY">

Wow, I’m really looking forward to it!!! You’re amazing!!!! 😭 I can’t wait to try it myself!!! 😊

**Comment by rolnight90** [comment]
> ## 👋 Bienvenue !
> Nous utilisons les Discussions comme lieu d'échange avec les autres membres de notre communauté. Nous espérons que vous :
> 
> * Posez les questions qui vous intriguent.
> * Partagez vos idées.
> * Interagissez avec les autres membres de la communauté.
> * Accueillez les autres et faites preuve d'ouverture d'esprit. N'oubliez pas que nous construisons cette communauté 
>   ensemble 💪.
> 
> Pour commencer, laissez un commentaire ci-dessous en vous présentant et en nous expliquant ce que vous faites au sein de cette communauté.

Bonjour moi c'est Laurent je suis un jeune guitariste de 50 ans et pour être honnête je suis aussi novice pour l'utilisation de footswitch midi et donc en fabriquer un est impensable vue mon niveau, 'j'ai donc voulu acheter un des appareils que vous vendez avec " pirate midi "seulement habitant en France je ne peux malheureusement pas me faire livré et donc commander un appareil 
Je voudrai donc savoir si vous auriez un autre moyen pour moi d'acquérir vos appareil déjà monté 
. J'aurais aimé contribuer à soutenir votre démarche qui aide beaucoup de guitariste.
Merci pour votre travail et j'espère que vous pourrez me permettre de commander un de vos appareils a bientôt cordialement ...

**Reply by Builty** [reply]
Hi. I've asked the Pirate Midi guys. They say they can ship to France now, and they have made contact with you.
Thanks.

**Comment by rolnight90** [comment]
Merci d'avoir pris de votre temps pour s'occuper de ma situation et
j'apprécie beaucoup votre geste, seulement malgré la réponse qui m'a été
faite de pirate midi, je n'arrive toujours pas a valider ma commande!
 je reçois un message:
 " changer d'adresse un produit dans votre panier ne permet pas la
livraison"
 De plus , le module "picot plus" que je voulais acheter n'est plus
disponible en stock!
C'est peut être un mal pour un bien!
Au final, Je me résigne à  effectuer la fabrication moi même en suivant vos
instructions.
Étant débutant en Diy et en MIDI
J'espère juste que j'y arriverai  .
Si je peux me permettre je profite de ce mail pour vous demander si vous
auriez un lien a me conseiller pour la fabrication avec l'écran tactile et
si possible avec la pédale mwave chocolaté que je possède déjà ?
A moins qu'il faille impérativement que je me procure un autre module
auxquels cas je le commanderai avec plaisir..
Encore merci pour votre investissement dans ce projet qui est vraiment un
plus pour tout les amoureux de la plateforme TONEX.
Je vous souhaite le meilleur pour l'avenir cordialement

**Comment by Builty** [comment]
Hi. The Pirate Midi guys tell me that everything should be OK for ordering from them now.
If you still do wish to make your own, refer here: https://github.com/Builty/TonexOneController/blob/main/HardwarePlatforms.md

These pages show the supported PCBs, and where to buy them. The Chocolate pedal can be used, but it needs the controller PCB also before it can work.

**Comment by rolnight90** [comment]
Bonjour, merci beaucoup pour les liens par contre je viens de visiter a
nouveau le site pirate midi et malheureusement les deux dual switch que
j'aurais aimé acheté sont en rupture de stock!
Dû coup si j'en veux un je n'ai pas d'autres choix que de le fabriquer.
Si je peux me permettre et si j'ai bien compris le contrôleur est sur le
circuit de l'écran je n'ai pas a acheté un contrôleur plus un écran ?
Promis après je vous ennuie plus cordialement....

**Comment by Builty** [comment]
Hi. Yes the PCB that runs the screen, also runs the Tonex control part. It's one part.

**Comment by rolnight90** [comment]
Bonjour Ok, thanks for the clarification.
Cordialement

**Comment by rolnight90** [comment]
Merci et désolé pour mon ignorance , encore une fois heureusement que
l'entraide existe encore c' est encourageant pour l'avenir.
Si je peux me permettre je voudrais pas abuser de votre gentillesse mais je
ne trouve pas le moyen d'acheter ou fabriquer la coque! pour par exemple
reproduire de a à z "le pirate max" (avec 2 footswitch et l'écran dans le
même boîtier.)
Y a t'il un lien avec toutes les étapes et la liste de tout les éléments a
acheter ou fabriquer? encore merci pour tout !
J'aurais sincèrement préféré acheter l'ensemble complet déjà monté !
 mais après avoir essayé de commander x fois  pendant 48h et pris contact
avec eux puis vous . Je n'arrive toujours pas a commander et cerise sur le
gâteau les 2 produits (polar mini et polar plus) ne sont plus en stock
😭😭😭
cela m'aurait permis d'acquérir ces pédales midi tout en contribuant a vous
soutenir.
Je m'excuse et vous souhaite le meilleur pour l'avenir cordialement

**Reply by Builty** [reply]
Hi.
Inside the Polar Max there is a custom PCB, and the outside case is also custom.

The details are here:
https://github.com/Pirate-MIDI/Polar/tree/main/Polar%20Max

However, it would need a 3D printer to make the case, and the PCB would be difficult for you to make.
If you really want to have the Polar Max, I think it would be better to keep in communication with the Pirate Midi guys and order one when its available.
